File: sdks/thunkmetrc-py/wrapper/src/thunkmetrc/wrapper/models/create_record_request.py
from typing import TypedDict, List, Optional, Any, Generic, TypeVar, TYPE_CHECKING
from typing_extensions import TypedDict

if TYPE_CHECKING:
    from . import *

class CreateRecordRequest(TypedDict, total=False):
    DocumentFileBase64: str
    DocumentFileName: str
    Label: str
    ResultDate: str
    Results: List[Any]

File: sdks/thunkmetrc-rs/wrapper/src/models/create_record_request.rs
use serde::{Deserialize, Serialize};
#[allow(unused_imports)]
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateRecordRequest {
    #[serde(rename = "DocumentFileBase64", skip_serializing_if = "Option::is_none")]
    pub document_file_base64: Option<String>,
    #[serde(rename = "DocumentFileName", skip_serializing_if = "Option::is_none")]
    pub document_file_name: Option<String>,
    #[serde(rename = "Label", skip_serializing_if = "Option::is_none")]
    pub label: Option<String>,
    #[serde(rename = "ResultDate", skip_serializing_if = "Option::is_none")]
    pub result_date: Option<String>,
    #[serde(rename = "Results", skip_serializing_if = "Option::is_none")]
    pub results: Option<Vec<serde_json::Value>>,
}
